### PR TITLE
TypeScript Make objects with object properties generic #9556

### PR DESCRIPTION
**Pull request by edloidas**

### CODE REVIEW

File: modules/lib/lib-auditlog/src/main/resources/lib/xp/auditlog.ts [hunk 1]
```diff
@@ -23,26 +23,26 @@ function checkRequired<T extends object>(obj: T, name: keyof T): void {
     }
 }
 
-export interface AuditLogParams {
+export interface AuditLogParams<DATA extends Record<string, unknown> = Record<string, unknown>> {
```
Comment (tajakobsen) [inline modules/lib/lib-auditlog/src/main/resources/lib/xp/auditlog.ts:26]: The convention used other places in the code is that type parameters are capitalized (not upper case).

```suggestion
export interface AuditLogParams<Data extends Record<string, unknown> = Record<string, unknown>> {
```

Comment (tajakobsen) [reply modules/lib/lib-auditlog/src/main/resources/lib/xp/auditlog.ts:26]: Where is it smart to have default type parameters (` = Record<string, unknown>`)? I'm thinking maybe on the functions, and not in the interfaces. Specially when the objects are only created by this library.

(This goes for the other cases below too.)

What do you think?

File: modules/lib/lib-project/src/main/resources/lib/xp/project.ts [hunk 2]
```diff
@@ -31,31 +31,31 @@ export interface ProjectReadAccess {
     public: boolean;
 }
 
-export interface CreateProjectParams {
+export interface CreateProjectParams<CONFIG extends Record<string, unknown> = Record<string, unknown>> {
```
Comment (tajakobsen) [inline modules/lib/lib-project/src/main/resources/lib/xp/project.ts:34]: The same comments as above. Don't uppercase type parameters, and think through where we want default type parameters (I think we maybe only want them on functions).

**Review by tajakobsen — CHANGES_REQUESTED** [review]
Good job. Think a bit about where it's smart to have *default type parameters*. I think it's maybe only on functions if the users are not creating instances of this object themselves.

**Review by ComLock — APPROVED** [review]
One problem with the current positional generics, is keeping backwards compatibility when one wants to add more generics later on.

Hopefully named generics will be added soon (give it a thumbs up?):
https://github.com/microsoft/TypeScript/issues/38913

**Comment by edloidas** [comment]
@tajakobsen Agree on all points. Had an uppercased template names in other project I worked on before this commit, so unintentionally used this style here too.
As for the default type params, we have an inconsistency across the project, so I looked through all the files and updated them: Params won't have defaults, but functions will. The result types, like AuditLog, Content, etc will also have a default type params in most of the cases.

**Review by tajakobsen — APPROVED** [review]
Looks good.

**Pull request merged**